src/configurationProvider.ts

```diff
@@ -510,27 +530,34 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 }
             }
 
-            const hintMessage = currentFile ?
-                `The file '${path.basename(currentFile)}' is not executable, please select a main class you want to run.` :
-                "Please select a main class you want to run.";
-            return this.promptMainClass(folder, progressReporter, hintMessage);
+            return this.promptMainClassUnderProject(config.projectName, progressReporter, "Please select a main class you wan to run");
         }
 
-        const containsExternalClasspaths = !_.isEmpty(config.classPaths) || !_.isEmpty(config.modulePaths);
-        const validationResponse = await lsPlugin.validateLaunchConfig(config.mainClass, config.projectName, containsExternalClasspaths, folder);
-        if (progressReporter.isCancelled()) {
-            return undefined;
-        } else if (!validationResponse.mainClass.isValid || !validationResponse.projectName.isValid) {
-            return this.fixMainClass(folder, config, validationResponse, progressReporter);
+        // Try to resolve main class from current file
+        const currentFile = config.mainClass || _.get(vscode.window.activeTextEditor, "document.uri.fsPath");
```

Comment:
We can simply use `window.activeTextEditor?.document.uri.fsPath` to avoid lodash here